hibernate-spatial/src/test/java/org/hibernate/spatial/integration/TestSpatialPredicates.java

```diff
@@ -0,0 +1,248 @@
+						30.0
+				);
+		retrieveAndCompare( dbexpected, predicateFactory );
+	}
+
+	@Test
+	public void isEmpty() throws SQLException {
+		if ( !isSupportedByDialect( SpatialFunction.isempty ) ) {
+			return;
+		}
+		Map<Integer, Boolean> dbexpected = expectationsFactory.getIsEmpty();
+		BiFunction<CriteriaBuilder, Root<JtsGeomEntity>, Predicate> predicateFactory = (criteriaBuilder, root) ->
+				SpatialPredicates.isEmpty( criteriaBuilder, root.get( "geom" ) );
+		retrieveAndCompare( dbexpected, predicateFactory );
+	}
+
+	@Test
+	public void isNotEmpty() throws SQLException {
+		if ( !isSupportedByDialect( SpatialFunction.isempty ) ) {
+			return;
+		}
+		Map<Integer, Boolean> dbexpected = expectationsFactory.getIsNotEmpty();
+		BiFunction<CriteriaBuilder, Root<JtsGeomEntity>, Predicate> predicateFactory = (criteriaBuilder, root) ->
+				SpatialPredicates.isNotEmpty( criteriaBuilder, root.get( "geom" ) );
+		retrieveAndCompare( dbexpected, predicateFactory );
+	}
+
+	@Test
+	public void havingSRID() throws SQLException {
+		if ( !isSupportedByDialect( SpatialFunction.srid ) ) {
+			return;
+		}
+		Map<Integer, Boolean> dbexpected = expectationsFactory.havingSRID( 4326 );
+		BiFunction<CriteriaBuilder, Root<JtsGeomEntity>, Predicate> predicateFactory = (criteriaBuilder, root) ->
+				SpatialPredicates.havingSRID( criteriaBuilder, root.get( "geom" ), 4326 );
+		retrieveAndCompare( dbexpected, predicateFactory );
+		dbexpected = expectationsFactory.havingSRID( 31370 );
+		predicateFactory = (criteriaBuilder, root) ->
+				SpatialPredicates.havingSRID( criteriaBuilder, root.get( "geom" ), 31370 );
+		retrieveAndCompare( dbexpected, predicateFactory );
+	}
+
+	private void retrieveAndCompare(
+			Map<Integer, Boolean> dbexpected,
+			BiFunction<CriteriaBuilder, Root<JtsGeomEntity>, Predicate> predicateFactory) {
+		Transaction tx = null;
+		try (Session session = openSession()) {
+			tx = session.beginTransaction();
+			CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+			CriteriaQuery<JtsGeomEntity> criteriaQuery = criteriaBuilder.createQuery( JtsGeomEntity.class );
+			Root<JtsGeomEntity> root = criteriaQuery.from( JtsGeomEntity.class );
+			criteriaQuery.select( root )
+					.where( predicateFactory.apply( criteriaBuilder, root ) );
+			List<JtsGeomEntity> list = session.createQuery( criteriaQuery )
+					.getResultList();
+			compare( dbexpected, list );
+		}
+		finally {
+			if ( tx != null ) {
+				tx.rollback();
+			}
+		}
+	}
+
+	private void compare(Map<Integer, Boolean> dbexpected, List<JtsGeomEntity> list) {
+		int cnt = 0;
+		for ( Map.Entry<Integer, Boolean> entry : dbexpected.entrySet() ) {
+			if ( entry.getValue() ) {
+				cnt++;
+				if ( !findInList( entry.getKey(), list ) ) {
+					fail( String.format( "Expected object with id= %d, but not found in result", entry.getKey() ) );
+				}
+			}
+		}
+		assertEquals( cnt, list.size() );
+		LOG.info( String.format( "Found %d objects within testsuite-suite polygon.", cnt ) );
```

Comment:
Seems we might use `LOG.infof("Found %d objects within testsuite-suite polygon.", cnt )` as well.

Reply:
This was also something I copied from `TestSpatialRestrictions`, but I guess I'll change it and also improve other parts of the tests that were copied from `TestSpatialRestrictions`